android/src/main/java/com/heapanalytics/reactnative/RNHeapLibraryModule.java

```
@@ -0,0 +1,110 @@
+
+  private final ReactApplicationContext reactContext;
+
+  public RNHeapLibraryModule(ReactApplicationContext reactContext) {
+    super(reactContext);
+    this.reactContext = reactContext;
+  }
+
+  @Override
+  public String getName() {
+    return "RNHeap";
+  }
+
+  @ReactMethod
+  public void setAppId(String appId) {
+    try {
+      Field skipInstrumentorCheckField = HeapImpl.class.getDeclaredField("skipInstrumentorChecks");
+      skipInstrumentorCheckField.setAccessible(true);
+      skipInstrumentorCheckField.setBoolean(null, true);
+    } catch (NoSuchFieldException e) {
+      System.out.println("Caught NoSuchFieldException when trying to skip the instrumentor checks");
+      e.printStackTrace();
+    } catch (IllegalAccessException e) {
+      System.out.println("Caught IllegalAccessException when trying to skip the instrumentor checks");
+      e.printStackTrace();
+    }
+
+    Heap.init(this.reactContext, appId);
+  }
+
+  @ReactMethod
+  public void identify(String identity) {
+    Heap.identify(identity);
+  }
+
+  private static Map<String, String> convertToStringMap(ReadableMap readableMap) {
+    Map<String, String> stringMap = new HashMap<>();
+    ReadableMapKeySetIterator mapIterator = readableMap.keySetIterator();
+    while (mapIterator.hasNextKey()) {
+      String key = mapIterator.nextKey();
+      switch (readableMap.getType(key)) {
+        case Null:
+          stringMap.put(key, null);
+          break;
+        case Number:
+          stringMap.put(key, "" + readableMap.getDouble(key));
+          break;
+        case Boolean:
+          stringMap.put(key, readableMap.getBoolean(key) ? "true" : "false");
+          break;
+        case String:
+          stringMap.put(key, readableMap.getString(key));
+          break;
+        case Array:
+          List<Object> list = readableMap.getArray(key).toArrayList();
+          List<String> stringList = new ArrayList<>();
+          for (Object o : list) {
```

As per our discussion, this transformation should throw an exception if the value is not a string, number, boolean, or null (as heap can't deal with arrays or maps).